gateway-sp-comms/src/single_sp.rs

```diff
@@ -752,6 +752,16 @@ impl SingleSp {
             response.expect_set_ipcc_key_lookup_value_ack()
         })
     }
+
+    pub async fn get_caboose_value(&self, key: [u8; 4]) -> Result<Vec<u8>> {
```

Comment:
Presumably this is returning a `Vec<u8>` to allow for arbitrary data without having to change the message type.